### DISCUSSION TITLE
signUp custom fields raw_user_meta_data

### DISCUSSION THREAD

**Discussion by selique** [supabase/supabase#7113] [Questions]
I try using add custom fields to signup data, but the server return "Database error saving new user ": 

*obs: I use Prisma like migration management because i don't use key foreign to create user, only trigger to copy id and data profile user to create data user profile
 
```sql
-- This trigger can be pasted into the first init migration
-- Copy from auth.users to public.users
DROP FUNCTION IF EXISTS handle_new_user cascade;
-- inserts a row into public.profiles table
create or replace function public.handle_new_user()
returns trigger as $$
    begin
        insert into public.profiles (
            id,
            full_name,
            avatar_url,
            nickname,
            matrial_status,
            gender,
            gender_indentity,
            cpf,
            birth_date,
            created_at
    ) values (
        new.id,
        new.raw_user_meta_data->>'full_name',
        new.raw_user_meta_data->>'avatar_url',
        new.raw_user_meta_data->>'nickname',
        new.raw_user_meta_data->>'matrial_status',
        new.raw_user_meta_data->>'gender',
        new.raw_user_meta_data->>'gender_indentity',
        new.raw_user_meta_data->>'cpf',
        new.raw_user_meta_data->>'birth_date',
        current_timestamp
        );
        return new;
    end;
$$ language plpgsql security definer set search_path = public;
```

**Comment by GaryAustin1** [comment]
Did you look in logs under dashboard database?  They might have more info. 
Do all your column names and types match?  I believe all the json will be text types. 
Try with just one column, the add in sets of columns until you get error.

**Comment by paulvonber** [comment]
@selique I guess you already found the answer, but if somebody else needs it:
`(new.raw_user_meta_data ->> 'birth_date'::TEXT)::DATE`

**Comment by xshadowlegendx** [comment]
hello, does this also works for http request? I tried this

```
{
	"email": "ddd@mail.com",
	"password": "123123123",
	"options": {
		"data": {
			"favorite": "yogurt"
		}
	}
}
```
but `new.raw_user_meta_data->>'favorite'` always return null and query the `raw_user_meta_data` field after user is signed up return empty json `{}`

nvm so remove the top level `options` and use only `data` field now it is working
